net/sockstats/sockstats_tsgo.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build tailscale_go && (unix || windows)
+
+package sockstats
+
+import (
+	"context"
+	"log"
+	"net"
+	"sync"
+	"sync/atomic"
+
+	"tailscale.com/net/interfaces"
+	"tailscale.com/wgengine/monitor"
+)
+
+type sockStatCounters struct {
+	txBytes, rxBytes                           atomic.Uint64
+	rxByBytesByInterface, txByBytesByInterface map[string]*atomic.Uint64
+}
+
+var (
```

Comment:
This is fine, but FYI, sometimes to make mutex-guarded values clearer I package these up in an anonymous type:

```
var counters struct {
    mu      sync.Mutex
    byLabel map[string]*sockStatCounters
    curInterface string
    ...
}
```

Then all the access sites are `counters.mu.Lock()`, `counters.byLabel`, etc.

Reply:
Added a `sockStats` anonymous struct (didn't call it `counters` because it has interfaces in there too)